protocol/x/prices/keeper/smoothed_price_test.go

```diff
@@ -38,15 +59,38 @@ func TestUpdateSmoothedPrices(t *testing.T) {
 				constants.MarketId2: constants.Exchange2_Price2_TimeT.Price + 35,
 				constants.MarketId7: constants.Price1,
 			},
+			linearInterpolateFunc: lib.Uint64LinearInterpolate,
 		},
 		"Initializes smoothed prices with index prices": {
-			indexPrices:    constants.AtTimeTSingleExchangePriceUpdate,
-			expectedResult: constants.AtTimeTSingleExchangeSmoothedPrices,
+			indexPrices:           constants.AtTimeTSingleExchangePriceUpdate,
+			expectedResult:        constants.AtTimeTSingleExchangeSmoothedPrices,
+			linearInterpolateFunc: lib.Uint64LinearInterpolate,
 		},
 		"All updated - multiple existing overlapped index and smoothed prices": {
-			indexPrices:    constants.AtTimeTSingleExchangePriceUpdate,
-			smoothedPrices: constants.AtTimeTSingleExchangeSmoothedPricesPlus10,
-			expectedResult: constants.AtTimeTSingleExchangeSmoothedPricesPlus7,
+			indexPrices:           constants.AtTimeTSingleExchangePriceUpdate,
+			smoothedPrices:        constants.AtTimeTSingleExchangeSmoothedPricesPlus10,
+			expectedResult:        constants.AtTimeTSingleExchangeSmoothedPricesPlus7,
+			linearInterpolateFunc: lib.Uint64LinearInterpolate,
+		},
+		"Interpolation errors - returns error": {
+			indexPrices:           constants.AtTimeTSingleExchangePriceUpdate,
+			smoothedPrices:        constants.AtTimeTSingleExchangeSmoothedPricesPlus10,
+			linearInterpolateFunc: errInterpolator,
+			expectedErr: "Error updating smoothed price for market 0: error while interpolating\n" +
+				"Error updating smoothed price for market 1: error while interpolating\n" +
+				"Error updating smoothed price for market 2: error while interpolating",
+			expectedResult: constants.AtTimeTSingleExchangeSmoothedPricesPlus10, // no change
+		},
+		"Single interpolation error - returns error, continues updating other markets": {
```

Comment:
nit: how about multiple interpolation errors?

Reply:
The above test case: `"Interpolation errors - returns error"` - returns multiple errors joined into a single error.